core/src/org/pentaho/di/core/Const.java

```diff
@@ -1357,6 +1357,30 @@ public static String rightPad( StringBuffer ret, int limit ) {
   }
 
   /**
+   * Right pad a StringBuilder: adds spaces to a string until a certain length. If the length is smaller then the limit
+   * specified, the String is truncated.
+   *
+   * @param ret
+   *          The StringBuilder to pad
+   * @param limit
+   *          The desired length of the padded string.
+   * @return The padded String.
+   */
+  public static String rightPad( StringBuilder ret, int limit ) {
```

Comment:
I am so sorry if I'm being dense here, but I don't follow why you need to do any mutation or instanceof checks.  These rightPad methods and several similar sets such as isEmpty() and stripCR() all have a very similar pattern:

They take some instance of CharSequence (and frequently a limit parameter), and then they use a for loop to iterate the sequence and build the return which is either a new String or a boolean.

Instead of having three methods for each of these, could we not just have one method and use the charAt(), length(), and subSequence() methods of CharSequence to implement those for loops?

I agree that other areas where we are mutating a buffer, and for those, we don't want to try to unify the builder version with it.